app/services/creator_service.py

```diff
@@ -230,3 +230,24 @@ async def post_async_contents_fetch(request_body: object, params: Optional[Dict]
         await executor_event.async_contents_fetch_request_event_handler(data=response_data)
 
     return response_data
+
+
+async def post_audience_overlap(request_body: object, params: Optional[Dict]) -> Optional[Dict]:
+    # Check if the product is supported
+    if Product.CREATOR_SEARCH not in settings.SUPPORTED_PRODUCTS:
+        raise HTTPException(
+            status_code=400,
+            detail=f"{Product.CREATOR_SEARCH} is not supported. \
+            Please add it to SUPPORTED_PRODUCTS in config to enable this API."
+        )
+
+    response_data: Dict = await post_audience_overlap_request(request_body=request_body, params=params)
```

Comment:
since python 3.7, we can use the native python dict type instead of typing's Dict
```suggestion
    response_data: dict = await post_audience_overlap_request(request_body=request_body, params=params)
```